lib/auth/oidc_test.go

```diff
@@ -74,10 +75,40 @@ func (s *OIDCSuite) SetUpSuite(c *check.C) {
 		SkipPeriodicOperations: true,
 	}
 	s.a, err = NewServer(authConfig)
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
+	return &s
 }
 
-func (s *OIDCSuite) TestCreateOIDCUser(c *check.C) {
+// createInsecureOIDCClient creates an insecure client for testing.
+func createInsecureOIDCClient(t *testing.T, connector types.OIDCConnector) *oidc.Client {
+	conf := oidcConfig(connector)
+	conf.HTTPClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+	client, err := oidc.NewClient(conf)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaults.WebHeadersTimeout)
+	defer cancel()
+
+	go func() {
+		defer cancel()
+		client.SyncProviderConfig(connector.GetIssuerURL())
```

Comment:
Can we drop the goroutine and do this in the body of `createInsecureOIDCClient`?